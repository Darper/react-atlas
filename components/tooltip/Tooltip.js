import React, { Component, PropTypes } from 'react';
import classNames from 'classnames/bind';
import style from './tooltip.css';

class Tooltip extends Component {
  render () {
    const {children, position, tooltip, inline, ...other} = this.props;

    const cx = classNames.bind(style);
    let tooltipClasses;
    if (!children.props.disabled) {
      tooltipClasses = cx({
        ["tooltip"]: true,
        ["tooltip-top"]: position !== 'left' && position !== 'bottom' && position !== 'right',
        ["tooltip-left"]: position === 'left',
        ["tooltip-bottom"]: position === 'bottom',
        ["tooltip-right"]: position === 'right'
      });
    }
    tooltipClasses = tooltipClasses ? tooltipClasses : '';

    const element = inline ? 'span' : 'div';

    const props = {
      ['data-tooltip']: tooltip,
      className: style.block + ' ' + tooltipClasses
    };

    return React.createElement(element, props, children);
  }
}

Tooltip.propTypes = {
  inline: PropTypes.bool,
  children: PropTypes.any,
  className: PropTypes.string,
  tooltip: PropTypes.string,
  tooltipDelay: PropTypes.number,
  tooltipHideOnClick: PropTypes.bool,
  position: PropTypes.string
};

Tooltip.defaultProps = {
  className: '',
  children: <p>This is some text with tooltip</p>,
  tooltip: 'Sample Tooltip Text'
};

export default Tooltip;
