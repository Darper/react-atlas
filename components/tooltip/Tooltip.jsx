import React from 'react';
import classNames from 'classnames/bind';
import style from './style.css';

const Tooltip = (ComposedComponent) => class extends React.Component {
  static propTypes = {
    children: React.PropTypes.any,
    className: React.PropTypes.string,
    onClick: React.PropTypes.func,
    onMouseEnter: React.PropTypes.func,
    onMouseLeave: React.PropTypes.func,
    tooltip: React.PropTypes.string,
    tooltipDelay: React.PropTypes.number,
    tooltipHideOnClick: React.PropTypes.bool
  };

  static defaultProps = {
    className: '',
    tooltipDelay: 0,
    tooltipHideOnClick: true
  };

  state = {
    active: false
  };

  handleMouseEnter = () => {
    if (this.timeout) clearTimeout(this.timeout);
    this.timeout = setTimeout(() =>this.setState({active: true}), this.props.tooltipDelay);
    if (this.props.onMouseEnter) this.props.onMouseEnter();
  };

  handleMouseLeave = () => {
    if (this.timeout) clearTimeout(this.timeout);
    if (this.state.active) this.setState({active: false});
    if (this.props.onMouseLeave) this.props.onMouseLeave();
  };

  handleClick = () => {
    if (this.timeout) clearTimeout(this.timeout);
    if (this.props.tooltipHideOnClick) this.setState({active: false});
    if (this.props.onClick) this.props.onClick();
  }

  render () {
    const {children, className, tooltip, tooltipDelay, tooltipHideOnClick, ...other} = this.props;
    const cx = classNames.bind(style);
    let composedClassName = cx({
      base: true,
      className: true
    });
    let tooltipClassName = cx({
      tooltip: true,
      active: this.state.active
    });

    return (
      <ComposedComponent
        {...other}
        className={composedClassName}
        onClick={this.handleClick}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        {children ? children : null}
        <span data-react-toolbox="tooltip" className={tooltipClassName}>{tooltip}</span>
      </ComposedComponent>
    );
  }
};

export default Tooltip;
