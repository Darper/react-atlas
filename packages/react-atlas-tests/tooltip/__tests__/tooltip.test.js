import React from "react";
import { mount } from "enzyme";
import { TooltipCore } from "../../../react-atlas-core/src/Tooltip/index";

describe("Testing TooltipCore component", () => {
	
	it("Tooltip component - Basic test", function() {});
	
});