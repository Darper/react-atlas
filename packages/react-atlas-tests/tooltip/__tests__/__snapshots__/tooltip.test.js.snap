// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component render Render correctly 1`] = `
<div
  className="class"
  data-tooltip=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={undefined}
  styleName="block tooltip tooltip-top"
>
  <i
    className="<i class='fa fa-github'></i>"
  />
  
</div>
`;
